### PR TITLE
feat: Make responsive to deal with smaller screens

### PR DESCRIPTION
**Pull request by askayastha22**
https://experience.adobe.com/?shell_source=dev&ref=genai-plugin-dev&repo=wknd&owner=hlxsites&project=WKND#/@sitesinternal/aem/generate-variations
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: web-src/src/components/App.js [hunk 1]
```diff
@@ -67,18 +72,44 @@ function NoAccessMessage() {
...
+  React.useEffect(() => {
+    const handleResize = () => {
+      setMainSidePanelState(window.innerWidth <= MAIN_SIDE_PANEL_COLLAPSE_WIDTH_THRESHOLD
+        ? MainSidePanelType.Collapsed
+        : MainSidePanelType.Expanded);
+      setMainSidePanelWidth(window.innerWidth <= MAIN_SIDE_PANEL_COLLAPSE_WIDTH_THRESHOLD
+        ? MAIN_SIDE_PANEL_COLLAPSE_WIDTH
+        : MAIN_SIDE_PANEL_EXPAND_WIDTH);
+    };
+    window.addEventListener('resize', handleResize);
```
Comment (basecode) [inline web-src/src/components/App.js:89]: Any chance to use CSS responsive media queries or is making react that impossible? `@media (min-width: 1600px) {`

**Review by basecode — APPROVED** [review]

File: web-src/src/components/App.js [hunk 2]
```diff
@@ -67,18 +72,44 @@ function NoAccessMessage() {
 
 export function App() {
   const [hasConsent, setConsent] = React.useState(true);
+  const [mainSidePanelWidth, setMainSidePanelWidth] = React.useState(MAIN_SIDE_PANEL_EXPAND_WIDTH);
```
Comment (vtsaplin) [inline web-src/src/components/App.js:75]: Static import?

Comment (vtsaplin) [inline web-src/src/components/App.js:75]: Static import?

File: web-src/src/components/App.js [hunk 3]
```diff
@@ -67,18 +72,44 @@ function NoAccessMessage() {
...
+        ? MAIN_SIDE_PANEL_COLLAPSE_WIDTH
+        : MAIN_SIDE_PANEL_EXPAND_WIDTH);
+    };
+    window.addEventListener('resize', handleResize);
+    handleResize();
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, []);
```
Comment (vtsaplin) [inline web-src/src/components/App.js:95]: Technically, it depends on setMainSidePanelWidth.

**Review by vtsaplin — APPROVED** [review]

File: web-src/src/components/App.js (hunk 1 above)
Comment (askayastha22) [reply web-src/src/components/App.js:89]: The panel sizes are set by the `columns` prop using Grid component's capabilities:
`<Grid columns={[mainSidePanelWidth, '1fr']} ...`

**Pull request merged**